src/common/snippets/include/snippets/lowered/pass/insert_specific_iterations.hpp

```diff
@@ -20,18 +25,67 @@ namespace pass {
 class InsertSpecificIterations : public RangedPass {
 public:
     OPENVINO_RTTI("InsertSpecificIterations", "RangedPass")
+    InsertSpecificIterations() = default;
     bool run(LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, lowered::LinearIR::constExprIt end) override;
 
     /**
-     * @brief Makes a copy of a loop body with id 'loop_id' and inserts it to the LinearIR before the 'insert_pos' position
-     * @param linear_ir LinearIR which should be modified
-     * @param loop_id id of the loop which should be copied
-     * @param insert_pos position before which the loop body copy should be inserted
-     * @return iterator which points on the LoopBegin copy
+     * @brief Check if specific Loop iterations needed
+     * @param unified_loop_info loop info of the original (unified) Loop
+     * @param type type of the specific loop iterations
+     * @param remaining_work_amount the work amount on the current moment (after applying of the previous loop decomposed parts)
+     * @return True if needed otherwise - False
+     */
+    static bool is_decomposed_loop_needed(const UnifiedLoopInfoPtr& unified_loop_info, SpecificLoopIterType type, size_t remaining_work_amount);
```

Comment:
Do we need all these methods as a part of public API?
I.E. can we move them to the `private`?